client/components/domains/domain-registration-suggestion/index.jsx

```diff
@@ -218,6 +231,27 @@ class DomainRegistrationSuggestion extends React.Component {
 		}
 	}
 
+	renderMatchReason() {
+		const { suggestion: { domain_name: domain }, isFeatured } = this.props;
+
+		if ( ! isFeatured || ! Array.isArray( this.props.suggestion.match_reasons ) ) {
```

Comment:
Isn't `! Array.isArray()` enforced by the PropTypes typecheck? I mean...It can't hurt to check again, but doesn't seem like it's needed.